lib/Context/Stack/Frame/index.ts

```diff
@@ -42,8 +42,10 @@ class Frame {
   public variables: Store;
 
   constructor(frameState: Options) {
-    this.nodeID = frameState.nodeID ?? frameState.blockID;
-    this.programID = frameState.programID ?? frameState.diagramID;
+    if (frameState.hasOwnProperty('blockID')) this.nodeID = frameState.blockID;
+    if (frameState.hasOwnProperty('nodeID')) this.nodeID = frameState.nodeID;
+
+    this.programID = frameState.diagramID ?? frameState.programID;
```

Comment:
`programID` is not a big deal because it is never expected to be falsy, a frame will ALWAYS have a programID